Assets/Scripts/UI/Canvas/ColorPicker/ColorPicker.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+using Framework.Enums;
+
+namespace UI.Canvas.ColorPicker
+{
+    public sealed class ColorPicker : MonoBehaviour
+    {
+        [SerializeField] private LineRenderer routeLineRenderer;
+        
+        private List<Color> colorList = new List<Color>()
+        {
+            Color.red,
+            Color.green,
+            Color.blue,
+            Color.yellow,
+            Color.magenta,
+            new Color(1f, 0.5f, 0f),
+            Color.black
+        };
+    
+        private void Start() => gameObject.SetActive(false);
+        
+        /// <summary>
+        /// Changes the line renderer's line color to a new color
+        /// </summary>
+        /// <param name="newColor"> The color to set the line to </param>
+        public void ChangeColor(ColorChange colorChange)
+        {
+            Color targetColor = GetColor(colorChange.colorToChange);
+            routeLineRenderer.startColor = targetColor;
+            routeLineRenderer.endColor = targetColor;
+        }
+        
+        private Color GetColor(RouteColors lineColor)
+        {
+            switch (lineColor)
+            {
+                case RouteColors.RED:
+                    return colorList[0];
+                case RouteColors.GREEN:
+                    return colorList[1];
+                case RouteColors.BLUE:
+                    return colorList[2];
+                case RouteColors.YELLOW:
+                    return colorList[3];
+                case RouteColors.MAGENTA:
+                    return colorList[4];
+                case RouteColors.ORANGE:
+                    return colorList[5];
+                case RouteColors.BLACK:
+                    return colorList[6];
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(lineColor), lineColor, null);
+            }
```

Comment:
```suggestion
            return colorList[(int)lineColor];
```
Als je een optionele feedback gebruik dan wel volledig toepassen.

Hoe dit werkt is simpel. Een Enum is uiteindelijk een lijst, dus je kan een index daarvan pakken. Als je zegt wordt getal en dat dan geeft aan die lijst zou het werken net als de switch case. Maar dan netter en korter, zelf vind ik dit meer DRY.

Reply:
```suggestion
            int index = (int)lineColor;
            if (index < 0
                    || index > colorList.Count)
                    return colorList[0];
           
           return colorList[index];
```
Dit is met checks want mogelijke bugs.